components/stacks-network/src/chains_coordinator.rs

```diff
@@ -380,30 +382,40 @@ pub async fn start_chains_coordinator(
                 };
                 let _ = devnet_event_tx.send(DevnetEvent::info(message));
 
-                let stacks_rpc = StacksRpc::new(&config.consolidated_stacks_rpc_url());
-                let _ = stacks_rpc.get_pox_info();
-
-                let should_submit_pox_orders = known_tip.block.metadata.pox_cycle_position
-                    == (known_tip.block.metadata.pox_cycle_length - 2);
-                if should_submit_pox_orders {
+                // only publish stacking order txs in tenure-change blocks
+                let has_coinbase_tx = known_tip
+                    .block
+                    .transactions
+                    .iter()
+                    .any(|tx| tx.metadata.kind == StacksTransactionKind::Coinbase);
+                if has_coinbase_tx {
                     let bitcoin_block_height = known_tip
                         .block
                         .metadata
                         .bitcoin_anchor_block_identifier
                         .index;
-                    let res = publish_stacking_orders(
-                        &config.devnet_config,
-                        &config.accounts,
-                        &config.services_map_hosts,
-                        config.deployment_fee_rate,
-                        bitcoin_block_height as u32,
-                    )
-                    .await;
-                    if let Some(tx_count) = res {
-                        let _ = devnet_event_tx.send(DevnetEvent::success(format!(
-                            "Will broadcast {} stacking orders",
-                            tx_count
-                        )));
+                    let should_submit_pox_orders = known_tip.block.metadata.pox_cycle_position
+                        == (known_tip.block.metadata.pox_cycle_length - 5);
```

Comment:
I think a comment explaining why the bool is calculated this way would be helpful